components/eam/src/chemistry/modal_aero/modal_aero_drydep.F90

```diff
@@ -733,4 +615,380 @@ subroutine calcram(ncol,landfrac,icefrac,ocnfrac, &
+  real(r8) function slip_correction_factor( dyn_visc, pres, temp, rair, pi, particle_radius ) 
+
+    real(r8),intent(in) :: dyn_visc         ! dynamic viscosity of air [kg m-1 s-1]
+    real(r8),intent(in) :: pres             ! air pressure [Pa]
+    real(r8),intent(in) :: temp             ! air temperature [K]
+    real(r8),intent(in) :: rair             ! gas constant of air [J/K/kg]
+    real(r8),intent(in) :: pi               ! constant pi = 3.14159....
+    real(r8),intent(in) :: particle_radius  ! particle radius [m]
+
+    real(r8) :: mean_free_path  ! [m]
+
+    mean_free_path = 2.0_r8*dyn_visc/( pres*sqrt( 8.0_r8/(pi*rair*temp) ) )
+
+    slip_correction_factor = 1.0_r8 + mean_free_path * &
+                             ( 1.257_r8+0.4_r8*exp(-1.1_r8*particle_radius/mean_free_path) ) / &
+                             particle_radius
+
+  end function slip_correction_factor
+
+  !====================================================================
+  ! Calculate the Schmidt number of air [unitless], see SeP97 p.972
+  !====================================================================
+  real(r8) function schmidt_number( temp, pres, radius, rair, pi, boltz, vsc_dyn_atm, vsc_knm_atm )
+
+    real(r8),intent(in) :: temp             ! air temperature [K]
+    real(r8),intent(in) :: pres             ! air pressure [Pa]
+    real(r8),intent(in) :: radius           ! particle radius [m]
+    real(r8),intent(in) :: rair             ! gas constant of air [J/K/kg]
+    real(r8),intent(in) :: pi               ! constant pi = 3.14159....
+    real(r8),intent(in) :: boltz            ! Boltzman's constant [J/K/molecule] 
+    real(r8),intent(in) :: vsc_dyn_atm      ! dynamic viscosity of air [kg m-1 s-1]
+    real(r8),intent(in) :: vsc_knm_atm      ! kinematic viscosity of air [m2 s-1]
+
+    real(r8) :: slp_crc   ! slip correction factor [unitless]
+    real(r8) :: dff_aer   ! Brownian diffusivity of particle [m2/s], see SeP97 p.474
+
+    slp_crc = slip_correction_factor( vsc_dyn_atm, pres, temp, rair, pi, radius )
+    dff_aer = boltz * temp * slp_crc / (6.0_r8*pi*vsc_dyn_atm*radius) 
+
+    schmidt_number = vsc_knm_atm / dff_aer
+
+  end function schmidt_number
+
+  !=======================================================================================
+  ! Calculate the bulk gravitational settling velocity [m/s-1] 
+  !  - using the terminal velocity of sphere falling in a fluid based on Stokes's law and 
+  !  - taking into account the influces of size distribution.
+  !=======================================================================================
+  real(r8) function gravit_settling_velocity( particle_radius, particle_density, gravit,       &
+                                              slip_correction, dynamic_viscosity, particle_sig )
+
+    real(r8),intent(in) :: particle_radius   ! [m]
+    real(r8),intent(in) :: particle_density  ! [kg/m3]
+    real(r8),intent(in) :: gravit            ! [kg/m2/s], gravitational acceleration
```

Comment:
Units should be [m/s^2]

Reply:
Good catch. This line has been removed when removing `gravit` from argument lists.